src/slacko.ml

```diff
@@ -335,6 +335,66 @@ type file_obj = {
   num_stars: int option [@default None];
 } [@@deriving of_yojson { strict = false }]
 
+type field_obj = {
+  title: string option [@default None];
+  value: string [@default ""];
+  short: bool option [@default None];
+} [@@deriving yojson { strict = false }]
+
+let field ?title ?short value = {
+  title;
+  value;
+  short;
+}
+
+type attachment_obj = {
+  fallback: string option [@default None];
+  color: string option [@default None];
+  pretext: string option [@default None];
+  author_name: string option [@default None];
+  author_link: string option [@default None];
+  author_icon: string option [@default None];
+  title: string option [@default None];
+  title_link: string option [@default None];
+  text: string option [@default None];
+  fields: field_obj list option [@default None];
+  image_url: string option [@default None];
+  thumb_url: string option [@default None];
+  footer: string option [@default None];
+  footer_icon: string option [@default None];
+  ts: timestamp option [@default None];
+  mrkdwn_in: string list option [@default None];
+} [@@deriving yojson { strict = false }]
+
+let ifnone a b = match a with Some v -> Some v | None -> b
+
+let attachment
+    ?fallback ?color ?pretext
+    ?author_name ?author_link ?author_icon
+    ?title ?title_link
+    ?text ?fields
+    ?image_url ?thumb_url
+    ?footer ?footer_icon
+    ?ts ?mrkdwn_in
+    () = {
+  fallback = ifnone fallback text;
+  color;
+  pretext;
+  author_name;
+  author_link;
+  author_icon;
+  title;
+  title_link;
+  text;
+  fields;
+  image_url;
+  thumb_url;
+  footer;
+  footer_icon;
+  ts;
+  mrkdwn_in;
```

Comment:
Rather than adding more parameters to `chat_post_message` (and potentially `chat_update`, which doesn't currently accept attachments, etc.) would it make sense to have a bigger `message` type that includes all the icons and attachments and whatnot instead of treating it as a string?

If so, I think that's a big enough change to be worth putting in its own PR after this one.